lang/java/avro/src/main/java/org/apache/avro/file/SnappyCodec.java

```diff
@@ -34,50 +35,63 @@ protected Codec createInstance() {
     }
   }
 
-  private SnappyCodec() {}
+  SnappyCodec() {}
 
   @Override public String getName() { return DataFileConstants.SNAPPY_CODEC; }
 
   @Override
   public ByteBuffer compress(ByteBuffer in) throws IOException {
```

Comment:
Tests were failing on only Snappy when I made some overly strict assumptions on the returned buffer. 

In the process of debugging I fixed at least one bug (not setting ByteOrder.LITTLE_ENDIAN and letting the file format depend on the CPU of the writer).

The code also was not properly accounting for arrayOffset in many cases.